tests/geometry/geodesic_test.py

```diff
@@ -71,6 +71,24 @@ def gt_geometry(
   return geometry.Geometry(cost_matrix=cost, kernel_matrix=kernel, epsilon=1.)
 
 
+def exact_heat_kernel(
+    G: Union[jnp.ndarray, nx.Graph], normalize: bool = False, t: float = 10
+):
+  L = jnp.diag(jnp.sum(G, axis=1)) - G
+  if normalize:
+    inv_sqrt_deg = jnp.diag(
+        jnp.where(
+            jnp.sum(G, axis=1) > 0.0, 1.0 / jnp.sqrt(jnp.sum(G, axis=1)), 0.0
```

Comment:
Can you please reformat, so that's a bit nicer? Maybe
```python
out_deg = jnp.sum(G, axis=1)
inv_sqrt_deg = jnp.where(out_deg > 0.0, 1.0 / out_deg, 0.0)
...
```